tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/MetadataIndexConstructor.java

```diff
@@ -62,49 +68,45 @@ public static MetadataIndexNode constructMetadataIndex(
       MetadataIndexNode currentIndexNode =
           new MetadataIndexNode(MetadataIndexNodeType.LEAF_MEASUREMENT);
       int serializedTimeseriesMetadataNum = 0;
+      String vectorName = ""; // record previous vector name
+      int numOfValueColumns = 0;
       for (int i = 0; i < entry.getValue().size(); i++) {
         timeseriesMetadata = entry.getValue().get(i);
-        if (timeseriesMetadata.isTimeColumn()) {
-          // calculate the number of value columns in this vector
-          int numOfValueColumns = 0;
+        if (numOfValueColumns > 0) {
+          // must be value column，不清楚这里是否需要加一层检验？外层可否保证？
```

Comment:
Good  job.Thanks for your contribution.
I have a little question.
generate , chinese is used in  notes